### PR TITLE
feat: Add SDK documentation field

### LINKED ISSUE #19: Separate field for sdk documentation
I was also wondering if we could have a separate field for url to the support documentation for sdk integration instructions since we all kinda throw them in either the short or long description field.  Perhaps sdk documentation/support url's warrant their own field?

_Originally posted by @jawz101 in https://github.com/Exodus-Privacy/etip/issues/11#issuecomment-472265850_

### PR DESCRIPTION
**Pull request by pnu-s**
Fixes #19 

~~https://github.com/Exodus-Privacy/etip/pull/118 needs to be merged first (because of migrations)~~

**Todo:**

- [x] Discuss field name `documentation` (possible choices: `documentation_links`, `sdk_documentation`, ?)
- [x] Discuss whether a comma-separated string field is OK
- [x] Validate whether 500 characters is enough
- [x] Open issue in exodus to add this field, get this data from ETIP endpoint and display it in tracker page 
- [x] Add proper tests for tracker creation with invalid documentation fields
- [ ] ~~Add documentation to `missing_fields` function?~~

We definitely need to take time to rewrite the tracker template `etip/trackers/templates/tracker.html`

@eighthave @jawz101 Feel free if you want to weigh in for the first 3 points :pray:

### CODE REVIEW

**Comment by jawz101** [comment]
I don't have issue with 500 character limit.

The comma separated thing - I don't know to what this is referring.  If it's in regards to signature strings- that's up to you all.  Personally, I like knowing and attempting to write the regex but there are some circumstances where there are some signatures that the Exodus team would write it more generally or more specifically, I may just start throwing a bunch of examples in a comment field and an Exodus team member could write the appropriate rule to create.

An example would be for Google, Facebook, or Amazon where they have multiple tracking products.  Personally, I just care that the company is involved but Exodus distinguishes each product individually.  In that case, I'd let them decide and perhaps the signature they would use would be more specific.

Basically, it's up to you all and it's probably a better system for it :)

**Comment by pnu-s** [comment]
@jawz101 I was just talking about the `documentation` field here, which is a newly added field for sdk documentation.

For instance it would be like: `https://link1.com,https://link2.com` with the current proposal

> Personally, I like knowing and attempting to write the regex but there are some circumstances where there are some signatures that the Exodus team would write it more generally or more specifically, I may just start throwing a bunch of examples in a comment field and an Exodus team member could write the appropriate rule to create.
> 
> An example would be for Google, Facebook, or Amazon where they have multiple tracking products. Personally, I just care that the company is involved but Exodus distinguishes each product individually. In that case, I'd let them decide and perhaps the signature they would use would be more specific.
> 
> Basically, it's up to you all and it's probably a better system for it :)

Thanks for this feedback anyway :)

**Comment by eighthave** [comment]
* I think comma-separated string field could be problematic for URLs  since 
commas are valid URLs.
* 500 Chars is probably OK for a single URL, but I think URLs are allowed to be 
up to 2000 chars each.
* It seems to me that an N-length list of string fields would best represent 
documentation.  Each entry would be a plain text URL.  Then just have a single 
field for all documentation links.
* Just `documentation` makes sense to me.

**Comment by pnu-s** [comment]
Thanks @eighthave !

> * I think comma-separated string field could be problematic for URLs since commas are valid URLs.

Damn, I did not think this through. I'm guessing a space-separated list could do the job, I'm not sure how handy it would be tough.

> * 500 Chars is probably OK for a single URL, but I think URLs are allowed to be up to 2000 chars each.

Django limits at 200 chars for a single URL (https://docs.djangoproject.com/en/2.2/ref/models/fields/#urlfield), so I'm thinking that 1000 for a list of URL should be sufficient (maybe 500 is a tight too small).

> * It seems to me that an N-length list of string fields would best represent documentation. Each entry would be a plain text URL. Then just have a single field for all documentation links.

For ETIP API, I think the most convenient way will be to return a list of strings anyway:

```
documentation: [
    "<link1>",
    "<link2>"
]
```

For ETIP & exodus implementation, I'm guessing we'll go for a space-separated list if no better idea comes up.

> * Just `documentation` makes sense to me.

Ok thanks!
